upup/pkg/fi/clientset_castore.go

```diff
@@ -610,15 +611,83 @@ func (c *ClientsetCAStore) DeleteSecret(item *KeystoreItem) error {
 	case SecretTypeKeypair:
 		client := c.clientset.Keysets(c.namespace)
 		return DeleteKeysetItem(client, item.Name, kops.SecretTypeKeypair, item.Id)
-
 	default:
 		// Primarily because we need to make sure users can recreate them!
 		return fmt.Errorf("deletion of keystore items of type %v not (yet) supported", item.Type)
 	}
 }
 
-// VFSPath implements CAStore::VFSPath
-func (c *ClientsetCAStore) VFSPath() vfs.Path {
-	// We will implement mirroring instead
-	panic("ClientsetCAStore::VFSPath not implemented")
+func (c *ClientsetCAStore) MirrorTo(basedir vfs.Path) error {
+	list, err := c.clientset.Keysets(c.namespace).List(v1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("error listing keysets: %v", err)
+	}
+
+	for i := range list.Items {
+		keyset := &list.Items[i]
+
+		if keyset.Spec.Type == kops.SecretTypeSecret {
+			continue
+		}
+
+		primary := FindPrimary(keyset)
+		if primary == nil {
+			glog.Warningf("skipping keyset with no primary data: %s", keyset.Name)
```

Comment:
Should we fail here? What is the use case?

Reply:
Good call - just being tolerant, but we don't expect this.  If we see it happening for some reason (it shouldn't), _then_ we can downgrade to a warning.